skimage/segmentation/_clear_border.py

```diff
@@ -1,37 +1,39 @@
 import numpy as np
-from scipy.ndimage import label
+#from scipy.ndimage import label
+from ..measure import label
 
 
-def clear_border(image, buffer_size=0, bgval=0):
-    """Clear objects connected to image border.
+def clear_border(labels, buffer_size=0, bgval=0):
+    """Clear objects connected to the label image border.
 
-    The changes will be applied to the input image.
+    The changes will be applied directly to the input.
 
     Parameters
     ----------
-    image : (N, M) array
-        Binary image.
+    labels : (N, M) array of int
+        Label or binary image.
     buffer_size : int, optional
-        Define additional buffer around image border.
+        The width of the border examined.  By default, only objects
+        that touch the outside of the image are removed.
     bgval : float or int, optional
-        Value for cleared objects.
+        Cleared objects are set to this value.
 
     Returns
     -------
-    image : (N, M) array
-        Cleared binary image.
+    labels : (N, M) array
+        Cleared binary image.  Note that the input label image is modified.
```

Comment:
I didn't like it either; I just documented it.

On Tue, Mar 24, 2015 at 5:04 PM, Juan Nunez-Iglesias <
notifications@github.com> wrote:

> In skimage/segmentation/_clear_border.py
> <https://github.com/scikit-image/scikit-image/pull/1449#discussion_r27085500>
> :
>
> >
> >      Returns
> >      -------
> > -    image : (N, M) array
> > -        Cleared binary image.
> > +    labels : (N, M) array
> > +        Cleared binary image.  Note that the input label image is modified.
>
> In general, I like the "functional" approach that scikit-image takes to
> most things. This breaks it. Another option is to use the out/output
> parameter, and work in-place when it is the same array as the input.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/scikit-image/scikit-image/pull/1449/files#r27085500>.
>